code-challenges.test.js

```diff
@@ -23,30 +39,86 @@ const secretCodeWord3 = "Eccentric"
+      expect(letter(filterletterA)).toEqual(["Mango", "Apricot", "Peach"])
+      expect(letter(filterletterB)).toEqual(["Cherry", "Blueberry", "Peach"])
+
+      const filterLetterA = "a"
+      // Expected output: ["Mango", "Apricot", "Peach"]
+      const filterLetterE = "e"
+      // Expected output: ["Cherry", "Blueberry", "Peach"]
+      const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    })
+  })
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+// pseudo code: 
+// input: array of words and a single letter
+// output/return: array of all the words containing that particular letter.
+// The letter function takes an array of words and a single letter. It returns a new array that only includes the words from the original array that have the specified letter. The function compares the lowercase versions of the words and the letter to make the comparison case-insensitive. The array now has the words that have the specified letter.
 
 // b) Create the function that makes the test pass.
 
+function letter(words, letter) {
+    return words.filter(word => word.toLowerCase().includes(letter.toLowerCase()));
+  }
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("cards", () => {
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+      expect(cards(hand1)).toEqual([true])
+      expect(cards(hand2)).toEqual([false])
+      expect(cards(hand3)).toEqual([false])
+      expect(cards(hand4)).toEqual([true])
+
+
+    })
+  })
+
+  const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        const hand4 = [7, 2, 7, 2, 7]
+        // Expecetd output: true
+
+// pseudo code: 
+// input: a function that takes in an array of 5 numbers
+// output/return: and array determining whether or not the array is a "full house"
+//   The cards function checks if a hand of cards represents a full house or not. It does this by counting the occurrences of each card in the hand and then checking if there are two cards with the same count (2 of a kind) and three cards with the same count (3 of a kind). If conditions are met, it returns true; if not then it returns false.
```

Comment:
You explained what the function does overall, but not how it does it in its execution.